src/BSON/ObjectID.c

```diff
@@ -67,15 +67,17 @@ PHP_METHOD(ObjectID, __construct)
 	zend_restore_error_handling(&error_handling TSRMLS_CC);
 
 	if (id) {
-		zend_str_tolower(id, id_len);
-		if (bson_oid_is_valid(id, id_len)) {
+		tid = zend_str_tolower_dup(id, id_len);
+		if (bson_oid_is_valid(tid, id_len)) {
 			bson_oid_t oid;
 
-			bson_oid_init_from_string(&oid, id);
+			bson_oid_init_from_string(&oid, tid);
 			bson_oid_to_string(&oid, intern->oid);
 		} else {
+			efree(tid);
```

Comment:
Is this a double free, given that you're doing it again a couple of lines below?